tspng/implantation.py

```diff
@@ -0,0 +1,82 @@
+# import statements
+import os
+import json
+
+from PIL import Image
+from PIL.PngImagePlugin import PngInfo
+from tspng import MIME_TYPE
+from typing import Dict
+
+
+def _implant_data(data: str, image: str, mime_type: str = MIME_TYPE, dst: str = None):
+    # open image
+    target_im = Image.open(image)
+    if target_im.format != "PNG":
+        raise Exception("Image is not a PNG.")
+    # implant data
+    metadata = PngInfo()
+    metadata.add_text(mime_type,data)
+    # save file with implanted data
+    mod_file = target_im.save(image[:-4]+'.ts.png', format="PNG", pnginfo=metadata)
```

Comment:
`image[:-4]` works, but all kinds of problems are going to occur. What if a TIFF file was passed with the extension, `.tiff`? Now, the `-4` is wrong.

I would use the [`splitext`](https://docs.python.org/3/library/os.path.html#os.path.splitext) function.

```python
base, _ = os.path.splitext(image)
base + ".ts.png"
```